### ISSUE TITLE
Rule proposal: `minimum-ternary`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1625]
### Description

The ternary part should be as small as possible.

### Fail

```js
const foo = test ? call(a) : call(b);
```

```js
const foo = test ? a(test) : b(test);
```

```js
const foo = test ? a + 1 : b + 1;
```

```js
const foo = test ? Math.min(a, 100) : Math.max(b, 100);
```

### Pass

```js
const foo = call(test ? a : b);
```

```js
const foo = (test ? a : b)(test);
```

```js
const foo = (test ? a : b) + 1;
```

```js
const foo = Math[test ? "min" : "max"](a, 100);
```

**Comment by fregante** [comment]
I don't know, that doesn't look always the best/clearest. Maybe 70/30. The last one definitely not great.

**Comment by fisker** [comment]
If the last is a call with complex arguments?

```js
	await (
		delayRejection
		 ? Promise.allSettled([
			  promise,
			  delay(minimumDelay)
		])
		: Promise.all([
			  promise,
			  delay(minimumDelay)
		])
	)
```

```js
	await Promise[delayRejection ? 'allSettled' : 'all']([
		promise,
		delay(minimumDelay)
	]);
```

https://github.com/sindresorhus/p-min-delay/blob/471c492464bea18cb7b13e24b3d62f2b403fb8e5/index.js#L4

**Comment by fregante** [comment]
That one yes, but also it looks like an obvious improvement one way or another. By obvious I mean that probably anyone looking at 4 identical lines will have the urge to DRY it 😃

**Comment by thernstig** [comment]
@sindresorhus @fisker honestly, I asked an entire team and everyone thought the originals looked better.

E.g. do you really think this looks better?

```js
const foo = (test ? a : b)();
```

Now the actual `()` block is far from the name of the function, so it is harder to see it invokes a function.

We are turning this off. I am not sure why it is recommended as our vote was unanimous.

**Comment by fregante** [comment]
For me it reported two instances:

- `isMac ? event.metaKey : event.ctrlKey`, which has already been reported and disabled by default in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/3222#issuecomment-4702945573
- `isMac ? call('Mac') : call('not-mac')`, which makes sense to reduce to `call(isMac ? 'Mac' : 'not-mac')`

So it has some merit but it needs to be less broad. It looks like we're in agreement that the ternary should not appear before a call.

**Comment by sindresorhus** [comment]
This was fixed in https://github.com/sindresorhus/eslint-plugin-unicorn/releases/tag/v67.0.0
